framework/src/modules/nft/internal_method.ts

```diff
@@ -82,20 +106,98 @@ export class InternalMethod extends BaseMethod {
 		});
 	}
 
-	public async transferInternal(
+	public async verifyTransfer(
+		immutableMethodContext: ImmutableMethodContext,
+		senderAddress: Buffer,
+		nftID: Buffer,
+	) {
+		const owner = await this._nftMethod.getNFTOwner(immutableMethodContext, nftID);
+
+		if (owner.length === LENGTH_CHAIN_ID) {
+			throw new TransferVerifyError(
+				'NFT is escrowed to another chain',
+				NftEventResult.RESULT_NFT_ESCROWED,
+			);
+		}
+
+		if (!owner.equals(senderAddress)) {
+			throw new TransferVerifyError(
+				'Transfer not initiated by the NFT owner',
+				NftEventResult.RESULT_INITIATED_BY_NONOWNER,
+			);
+		}
+
+		const lockingModule = await this._nftMethod.getLockingModule(immutableMethodContext, nftID);
+
+		if (lockingModule !== NFT_NOT_LOCKED) {
+			throw new TransferVerifyError(
+				'Locked NFTs cannot be transferred',
+				NftEventResult.RESULT_NFT_LOCKED,
+			);
+		}
+	}
+
+	public async verifyTransferCrossChain(
+		immutableMethodContext: ImmutableMethodContext,
+		senderAddress: Buffer,
+		nftID: Buffer,
+		sendingChainID: Buffer,
+		receivingChainID: Buffer,
+		messageFee: bigint,
+		data: string,
+	) {
+		if (receivingChainID.equals(sendingChainID)) {
+			throw new TransferVerifyError(
+				'Receiving chain cannot be the sending chain',
+				NftEventResult.INVALID_RECEIVING_CHAIN,
+			);
+		}
+
+		// perform checks that are common for same-chain and cross-chain transfers
+		await this.verifyTransfer(immutableMethodContext, senderAddress, nftID);
```

Comment:
it seems like this function is called from outside in the transfer cross chain command

Reply:
Good catch 👍🏻